torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -771,12 +773,12 @@ void ProcessGroupAgent::pollTimedOutRPCs() {
         --clientActiveCalls_;
```

Comment:
@jjlilley @osalpekar Would any of you know when this condition potentially does not trigger? It seems like this should be the first time we are processing the future, as even if a receive and timeout arrive at the same time, we grab the lock to process both, so only one should process successfully

Reply:
Can we get in the situation where the future is satisfied by regular IO, when the lock is dropped, racing with this logic?
If anything, we should probably check !complete(). Or call setErrorIfNeeded() rather than setError().

(aside: ideally we should modify future<T> api to return a bool from setErrorIfNeeded(), so that we can make adjust logic if it was needed)